### PR TITLE
Fix host name, debuggability and embed CRT

### LINKED ISSUE #835: CLIHost should statically link CRT for Windows
https://github.com/dotnet/cli/issues/195#issuecomment-171169228

### PR DESCRIPTION
**Pull request by schellap**
Fixes #835

Links CRT into the DLL as well.

Also, as the host gets too complex it would be nice to debug it :smile:

@piotrpMSFT you will thank me after this PR (i.e., you can set breakpoints.)
@gkhanna79 PTAL.

/Cc @anurse

### CODE REVIEW

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu Debug Build please.

**Comment by piotrpMSFT** [comment]
@schellap thanks!!

**Comment by piotrpMSFT** [comment]
:shipit:

File: src/corehost/cli/dll/CMakeLists.txt [hunk 1]
```diff
@@ -26,5 +34,5 @@ endif()
 
 add_definitions(-DCOREHOST_MAKE_DLL=1)
 
-add_library(clihost SHARED ${SOURCES})
+add_library(hostimpl SHARED ${SOURCES})
```
Comment (anurse) [inline src/corehost/cli/dll/CMakeLists.txt]: Can I suggest a minor change to `corehostimpl`, `coreclrhost`, `clrhostimpl`, etc.? Basically just to make it clear that this is related to the CLR hosting specifically. As a developer, I think that having a random dylib inserted into my build output would be a little surprising to me but if it was named something that made it clear it was used to launch the CLR that would reassure me :)

**Comment by anurse** [comment]
Looks good! :shipit: after a brief consideration on the name.

**Comment by gkhanna79** [comment]
LGTM

**Comment by schellap** [comment]
@dotnet-bot test Ubuntu Debug Build please

**Comment by schellap** [comment]
@dotnet-bot test OSX Debug Build please

**Pull request merged**